app.py

```diff
@@ -17,9 +17,12 @@
 
 st.set_page_config(page_title="Student Name Detector (English)", layout="wide")
 
+# Pickled vectorizer freaks out if this isn't here
+def identity(x):
```

Comment:
lol. I can't imagine why this is necessary, but whatever...

Reply:
Here's a reference:
https://rebeccabilbro.github.io/module-main-has-no-attribute/

Seems like a more fundamental fix would involve retraining or repackaging the model, and I didn't really want to spend time looking into this.